src/Log/SentryLogger.php

```diff
@@ -332,4 +333,80 @@ public static function backtrace(array $record) : array
         ]);
     }
 
+    /**
+     * Get the information about the current release, if possible
+     * @return array
+     */
+    protected function getGitInfo()
+    {
+        // Initialise empty array to return
+        $return = [];
+        // If the file operations error out, we need to catch it
+        try {
+            $return = $this->getGitCommitMessage($return);
+        } catch (\Exception $exception) {
+            // Default message. As it's an extra, it's not overly crowding the interface
+            $return['Message'] = 'No git repo found or inaccessible';
+        }
+
+        return $return;
+    }
+
+    /**
+     * Get the HEAD tag or the release
+     *
+     * @return string
+     */
+    public static function getGitHead()
+    {
+        $head = file_exists(Director::baseFolder() . '/.git/HEAD');
+        if ($head) {
+            $head = explode(':', file_get_contents(Director::baseFolder() . '/.git/HEAD'));
+            $ref = file_get_contents(Director::baseFolder() . '/.git/' . trim($head[1]));
+            $ref = self::getGitTag($ref);
+            return $ref;
+        }
+
+        return '';
+    }
+
+    /**
+     * Check if we are on a tagged release and if so, use that to set the client release
+     * @param $ref
+     * @return mixed
+     */
+    protected static function getGitTag($ref)
+    {
+        $data = file_get_contents(Director::baseFolder() . './git/packed-refs');
+        if ($data) {
+            $data = explode('\n', $data);
+            foreach ($data as $line) {
+                if (strpos($line, '#') !== false) {
+                    continue;
+                }
+                $tags = explode(' ', $line);
+                if ($tags[0] === $ref) {
+                    list(,,$tag) = explode('/', $ref[1]);
+                    return $tag;
+                }
+            }
+        }
+
+        return $ref;
+    }
+
+    /**
+     * @param array $return
+     * @return array
+     */
+    protected function getGitCommitMessage(array $return): array
+    {
+        $msgFile = file_exists(Director::baseFolder() . '/.git/COMMIT_EDITMSG');
```

Comment:
Not a massive deal, but if logic needs to make so many F/S calls to achieve its aims, why not do it once for each required file, and cache the result as an array of your selected files, to the current scope?